wandb/sdk/launch/sweeps/scheduler_sweep.py

```diff
@@ -0,0 +1,145 @@
+        while True:
+            if not self.is_alive():
+                return
+            # AgentHeartbeat wants dict of runs which are running or queued
+            _run_states = {}
+            for run_id, run in self._runs.items():
+                if run.state == SimpleRunState.ALIVE:
+                    _run_states[run_id] = True
+            _msg = f"AgentHeartbeat sending: \n{pprint.pformat(_run_states)}\n"
+            logger.debug(_msg)
+            wandb.termlog(_msg)  # TODO: Remove
+            commands = self._api.agent_heartbeat(
+                self._heartbeat_agent_id, {}, _run_states
+            )
+            if commands:
+                _msg = f"AgentHeartbeat received {len(commands)} commands: \n{pprint.pformat(commands)}\n"
+                logger.debug(_msg)
+                wandb.termlog(_msg)  # TODO: Remove
+                for command in commands:
+                    _type = command.get("type")
+                    # type can be one of "run", "resume", "stop", "exit"
+                    if _type == "exit":
+                        self.state = SchedulerState.COMPLETED
+                        self.exit()
+                        return
+                    run = SweepRun(
+                        id=command.get("run_id"),
+                        args=command.get("args"),
+                        logs=command.get("logs"),
+                        program=command.get("program"),
+                    )
+                    self._runs[run.id] = run
+                    if _type in ["run", "resume"]:
+                        self._heartbeat_queue.put(run)
+                    elif _type == "stop":
+                        self._stop_run(run.id)
+                        continue
+            time.sleep(self._heartbeat_thread_sleep)
+
+    def _run(self) -> None:
+        try:
+            run: SweepRun = self._heartbeat_queue.get(
+                timeout=self._heartbeat_queue_timeout
+            )
+        except queue.Empty:
+            _msg = "No jobs in Sweeps RunQueue, waiting..."
+            logger.debug(_msg)
+            wandb.termlog(_msg)
+            time.sleep(self._main_thread_sleep)
+            return
+        # If run is already stopped just ignore the request
+        if run.state in [
+            SimpleRunState.DEAD,
+            SimpleRunState.UNKNOWN,
+        ]:
+            return
+        _msg = f"Converting Sweep Run (RunID:{run.id}) to Launch Job"
+        logger.debug(_msg)
+        wandb.termlog(_msg)
+        # This is actually what populates the wandb config
+        # since it is used in wandb.init()
+        sweep_param_path = os.path.join(
+            os.environ.get(wandb.env.DIR, os.getcwd()),
+            "wandb",
+            f"sweep-{self._sweep_id}",
+            f"config-{run.id}.yaml",
+        )
+        wandb.termlog(f"Saving params to {sweep_param_path}")
+        wandb_lib.config_util.save_config_file_from_dict(sweep_param_path, run.args)
```

Comment:
We will eventually need to move the sweep config stuff to an environment variable. Since we're running with launch, this should just work right now, but will need to be done when we move away from local_process

Reply:
All you should need to do is put the config here into the override configs in the add to launch queue call below

Reply:
Okay, I piped it through `params` which seems to get merged into override configs